File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Events/OnHealAfterMessage.ts
import { IMessage } from "../../../../Core/ObserverMessageKit";
import AbstractCreature from "../Unit/AbstractCreature";

export class OnHealAfterMessage implements IMessage{
    public static readonly Id = "OnHealAfterMessage";
    public constructor(public source:AbstractCreature,public Amount : number){

    }
}

File: Unity/Assets/Scripts/ModTheSpire/Event/Ids/OnHealAfterMessage.cs
using ModTheSpire.Unit;

namespace ModTheSpire.Event.Ids
{
    ///受到治疗前
    public class OnHealAfterMessage : IEventHandle
    {
        public AbstractCreature Unit;
        public int Amount;

        public OnHealAfterMessage(AbstractCreature unit, int amount)
        {
            Unit = unit;
            Amount = amount;
        }
    }
}